eventhubs/client/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubConsumerOptions.java

```diff
@@ -0,0 +1,219 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.messaging.eventhubs;
+
+import com.azure.core.amqp.Retry;
+import com.azure.core.implementation.util.ImplUtils;
+import reactor.core.scheduler.Scheduler;
+
+import java.time.Duration;
+import java.util.Locale;
+import java.util.Optional;
+
+/**
+ * Options when receiving events from Event Hubs.
+ */
+public class EventHubConsumerOptions implements Cloneable {
+
+    /**
+     * The maximum length, in characters, for the identifier assigned to an {@link EventHubConsumer}.
+     */
+    public static final int MAXIMUM_IDENTIFIER_LENGTH = 64;
+    /**
+     * The minimum value allowed for the prefetch count of the consumer.
+     */
+    public static final int MINIMUM_PREFETCH_COUNT = 1;
+    /**
+     * The maximum value allowed for the prefetch count of the consumer.
+     */
+    public static final int MAXIMUM_PREFETCH_COUNT = 8000;
+
+    // Default number of events to fetch when creating the consumer.
+    static final int DEFAULT_PREFETCH_COUNT = 500;
+
+    private String identifier;
+    private Long ownerLevel;
+    private Retry retryPolicy;
+    private Scheduler scheduler;
+    private int prefetchCount;
+
+    /**
+     * Creates a new instance with a position at the beginning of the partition stream, and the default prefetch amount.
+     */
+    public EventHubConsumerOptions() {
+        this.prefetchCount = DEFAULT_PREFETCH_COUNT;
+    }
```

Comment:
You say this method starts at beginning of partition stream, but I don't see that getting set anywhere? Maybe it is implicit but just double-checking.

Reply:
This is remnants of the previous API from 2 days ago, before API change said that event position should be mandatory when creating a consumer